sdk/identity/azure-identity/src/main/java/com/azure/identity/ChainedTokenCredential.java

```diff
@@ -36,18 +38,29 @@
     @Override
     public Mono<AccessToken> getToken(TokenRequestContext request) {
         AtomicReference<Throwable> cause = new AtomicReference<>();
+        List<CredentialUnavailableException> exceptions = new ArrayList<>(4);
         return Flux.fromIterable(credentials)
-            .flatMap(p -> p.getToken(request).onErrorResume(t -> {
-                if (cause.get() != null) {
-                    t.initCause(cause.get());
-                }
-                cause.set(t);
-                return Mono.empty();
-            }), 1)
-            .next()
-            .switchIfEmpty(Mono.defer(() -> Mono.error(new RuntimeException("Tried "
-                + credentials.stream().map(c -> c.getClass().getSimpleName()).collect(Collectors.joining(", "))
-                + " but failed to acquire a token for any of them. Please verify the environment for either of them"
-                + " and see more details in the causes below.", cause.get()))));
+                   .flatMap(p -> p.getToken(request).onErrorResume(CredentialUnavailableException.class, t -> {
+                       exceptions.add(t);
+                       return Mono.empty();
+                   }), 1)
```

Comment:
nit: don't need to specify concurrency as 1 for flatmap. It's 1 by default.

Reply:
That's not true.
Default concurrency is computed via:
Math.max(16, Integer.parseInt(System.getProperty("reactor.bufferSize.small", "256")));